docs/reference/files/charmcraft-yaml-file.rst

```diff
@@ -424,6 +432,46 @@ secret URI.
         :start-at: config:
         :end-before: containers:
 
+.. admonition:: Best Practice
+    :class: hint
+
+    Charms should keep the user experience of the operator as simple and obvious
+    as possible. Aim to make the application deployable without providing any
+    further configuration options: ``juju deploy foo`` is preferable to
+    ``juju deploy foo --config required-option=value``.
+
+.. admonition:: Best Practice
+    :class: hint
+
+    Consider which of your application's configuration options you should
+    initially expose. If your application has many config options, it may be
+    prudent to provide access to a select few, and add support for more as the
+    need arises.
```

Comment:
It feels like the intent here is to say: Be thrifty with how many configs = knobs you expose. Just like Juju, a charm is supposed to be an opinionated tool. I'd rephrase something like that. (Because "initially expose" makes me wonder: Is it okay to end up a gazillion configs later on?)